### PR TITLE
PTEUDO-1616: update dsns when a new role claim is created

### PR DESCRIPTION
**Pull request by leandrorichardtoledo**
- https://infoblox.atlassian.net/browse/PTEUDO-1616

### CODE REVIEW

File: pkg/roleclaim/roleclaim.go [hunk 1]
```diff
@@ -585,24 +581,53 @@ func (r *DbRoleClaimReconciler) copySourceSecret(ctx context.Context, sourceSecr
 	sourceSecretData := sourceSecret.Data
 
 	if newUser != "" {
+		oldUser := string(sourceSecretData["username"])
```
Comment (drewwells) [inline pkg/roleclaim/roleclaim.go:584]: This is a lot of duplication. How about
```
if newUser != "" || newPassword != "" {
```

File: pkg/roleclaim/roleclaim.go [hunk 2]
```diff
@@ -585,24 +581,53 @@ func (r *DbRoleClaimReconciler) copySourceSecret(ctx context.Context, sourceSecr
...
+
 	if newPassword != "" {
+		oldPassword := string(sourceSecretData["password"])
+
+		dsn := string(sourceSecretData["dsn.txt"])
+		dsn = strings.Replace(dsn, "password="+oldPassword, "password="+newPassword, 1)
+		sourceSecretData["dsn.txt"] = []byte(dsn)
+
+		roDsn := string(sourceSecretData["ro_uri_dsn.txt"])
+		roDsn = strings.Replace(roDsn, oldPassword, newPassword, 1)
```
Comment (drewwells) [inline pkg/roleclaim/roleclaim.go:609]: You have to escape the passwords so they're URL safe. Please use the dbclient methods for populating DSN as they're verified to work.

https://github.com/infobloxopen/db-controller/blob/main/pkg/databaseclaim/secrets.go#L33-L34

**Review by drewwells — CHANGES_REQUESTED** [review]
passwords need url escaping

File: pkg/roleclaim/roleclaim.go (hunk 2 above)
Comment (leandrorichardtoledo) [reply pkg/roleclaim/roleclaim.go]: Refactored.

File: pkg/roleclaim/roleclaim.go [hunk 3]
```diff
@@ -587,22 +583,36 @@ func (r *DbRoleClaimReconciler) copySourceSecret(ctx context.Context, sourceSecr
...
+	database := string(sourceSecretData["database"])
+	hostname := string(sourceSecretData["hostname"])
+	password := string(sourceSecretData["password"])
+	port := string(sourceSecretData["port"])
+	sslmode := string(sourceSecretData["sslmode"])
+	username := string(sourceSecretData["username"])
+
+	sourceSecretData["dsn.txt"] = []byte(dbclient.PostgresConnectionString(hostname, port, username, password, database, sslmode))
+	sourceSecretData["uri_dsn.txt"] = []byte(dbclient.PostgresURI(hostname, port, username, password, database, sslmode))
+	if sourceSecretData["ro_uri_dsn.txt"] != nil {
```
Comment (drewwells) [inline pkg/roleclaim/roleclaim.go:600]: I think default of []byte is nil, so this is always going to be nil

File: pkg/roleclaim/roleclaim.go [hunk 4]
```diff
@@ -587,22 +583,36 @@ func (r *DbRoleClaimReconciler) copySourceSecret(ctx context.Context, sourceSecr
...
+		sourceSecretData["ro_uri_dsn.txt"] = []byte(dbclient.PostgresURI(hostname, port, username, password, database, sslmode))
```
Comment (drewwells) [inline pkg/roleclaim/roleclaim.go:601]: `ro_uri_dsn.txt` is different than `uri_dsn.txt` 
https://github.com/infobloxopen/db-controller/blob/main/pkg/databaseclaim/secrets.go#L37

**Review by drewwells — COMMENTED** [review]
ro dsn needs a little work

**Pull request merged**